### PR TITLE
Issue #306 - Typecast error in SAML2DefaultResponseValidator

### PR DESCRIPTION
**Pull request by jkacer**
Fixed a possible NPE after the previous fix.

### CODE REVIEW

File: pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2DefaultResponseValidator.java [hunk 1]
```diff
@@ -649,9 +649,12 @@ protected final void validateAssertionSignature(final Signature signature, final
         if (signature != null) {
             final String entityId = peerContext.getEntityId();
             validateSignature(signature, entityId, engine);
-        } else if (context.getSPSSODescriptor().getWantAssertionsSignedXSBoolean().getValue()
-                && !peerContext.isAuthenticated()) {
-            throw new SAMLException("Assertion or response must be signed");
+        } else {
+            SPSSODescriptor spDescriptor = (context == null) ? null : context.getSPSSODescriptor();
+            Boolean wantAssertionsSigned = (spDescriptor == null) ? Boolean.FALSE : spDescriptor.getWantAssertionsSigned();
```
Comment (leleuj) [inline pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2DefaultResponseValidator.java]: Why `Boolean` and not `boolean`?

**Comment by leleuj** [comment]
Expecting @mmoayyed's review before the merge...

File: pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2DefaultResponseValidator.java (hunk 1 above)
Comment (jkacer) [inline pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2DefaultResponseValidator.java]: The library method returns a Boolean, so I kept the type.

Comment (leleuj) [inline pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2DefaultResponseValidator.java]: Understood

File: pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2DefaultResponseValidator.java [hunk 2]
```diff
@@ -649,9 +649,12 @@ protected final void validateAssertionSignature(final Signature signature, final
         if (signature != null) {
             final String entityId = peerContext.getEntityId();
             validateSignature(signature, entityId, engine);
-        } else if (context.getSPSSODescriptor().getWantAssertionsSignedXSBoolean().getValue()
-                && !peerContext.isAuthenticated()) {
-            throw new SAMLException("Assertion or response must be signed");
+        } else {
+            SPSSODescriptor spDescriptor = (context == null) ? null : context.getSPSSODescriptor();
```
Comment (mmoayyed) [inline pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2DefaultResponseValidator.java]: Can `context` ever be null? I think this check is unneeded.

**Comment by mmoayyed** [comment]
Reviewed. Looks fine, just a comment on a condition added.

File: pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2DefaultResponseValidator.java (hunk 2 above)
Comment (jkacer) [inline pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2DefaultResponseValidator.java]: Probably not, but I was not sure.

**Pull request merged**